### PR TITLE
controlling member validation + captcha ep

### PR DESCRIPTION
**Pull request by esdd1995**
# Description

This PR includes the following proposed change(s):

-[SPDBT-2927] request validation
-[SPDBT-2929] captcha

### CODE REVIEW

File: src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            .When(r => r.ResidentialAddress != null);
+        RuleFor(r => r.HasCriminalHistory).NotEmpty();
+        RuleFor(r => r.CriminalHistoryDetail)
+            .NotEmpty()
+            .MaximumLength(1000)
+            .When(r => r.HasCriminalHistory == true);
+        RuleFor(r => r.HasBankruptcyHistory).NotEmpty();
+        RuleFor(r => r.BankruptcyHistoryDetail)
+            .NotEmpty()
+            .MaximumLength(1000)
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs:45]: Maximum in confluence is 250 chars

File: src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        RuleFor(r => r.ResidentialAddress.Country).NotEmpty()
+            .MaximumLength(100)
+            .When(r => r.ResidentialAddress != null);
+        RuleFor(r => r.ResidentialAddress.PostalCode).NotEmpty()
+            .MaximumLength(20)
+            .When(r => r.ResidentialAddress != null);
+        RuleFor(r => r.HasCriminalHistory).NotEmpty();
+        RuleFor(r => r.CriminalHistoryDetail)
+            .NotEmpty()
+            .MaximumLength(1000)
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs:40]: Maximum in confluence is 250 chars

File: src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        RuleFor(r => r.OtherOfficerRole)
+            .NotEmpty()
+            .MaximumLength(50)
+            .When(r => r.IsPoliceOrPeaceOfficer == true && r.PoliceOfficerRoleCode == PoliceOfficerRoleCode.Other);
+
+        RuleFor(r => r.Aliases)
+            .NotEmpty()
+            .Must(r => r.Count() <= Constants.MaximumNumberOfUserEnteredAliases)
+            .When(r => r.HasPreviousNames == true)
+            .WithMessage("No more than 10 user entered aliases are allowed");
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs:61]: Maximum specified in confluence is 3 and all of the name fields have a max length of 40 chars.

File: src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                aliases.RuleFor(alias => alias.Surname)
+                    .NotEmpty()
+                    .WithMessage("Surname name is required for all aliases.");
+            })
+            .When(r => r.Aliases != null && r.HasPreviousNames == true);
+
+        RuleFor(r => r.IsTreatedForMHC).NotEmpty();
+        RuleFor(r => r.IsCanadianCitizen).NotEmpty();
+        RuleFor(r => r.BcDriversLicenceNumber)
+            .MaximumLength(7)
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs:75]: BcDriversLicenceNumber must be 7 or 8 digits

File: src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Spd.Manager.Licence;
+public class ControllingMemberCrcAppValidator : AbstractValidator<ControllingMemberCrcAppSubmitRequest>
+{
+    public ControllingMemberCrcAppValidator()
+    {
+        RuleFor(r => r.ParentBizLicApplicationId).NotEmpty();
+        RuleFor(r => r.Surname).NotEmpty();
```
Comment (carolcarpenter) [inline src/Spd.Manager.Licence/ControllingMemberCrcAppValidation.cs:15]: Name fields have a max length of 40

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**